snake.py

```diff
@@ -0,0 +1,227 @@
+    pygame.display.update()
+    
+    running = True
+    while running:
+        for event in pygame.event.get():
+            if event.type == KEYDOWN:
+                if event.key == K_y : 
+                    running = False
+
+                    ### Adds player(s)
+                    p1 = Player((360,240),STAY)
+                    p2 = Player((280,240),STAY)
+                    p2.body_color = GREEN
+                    p2.head_color = DARK_GREEN
+
+                    ### Position of fruit
+                    x_fruit = random.randrange(20, 620, 20)
+                    y_fruit = random.randrange(20, 460, 20)
+                    fruit = (x_fruit, y_fruit)
+
+                    ### Set game speed
+                    speed = 3
+
+                if event.key == K_n :
+                    print('game ended')
+                    pygame.quit()
+                    sys.exit()
+                    pygame.display.update()
+
+            # Quit if window closed
+            if event.type == QUIT:
+                print('game ended')
+                pygame.quit()
+                sys.exit()
+                pygame.display.update()
+
+### Screen
+screen_size = (640, 480)
+screen_center = (int(screen_size[0]/2),int(screen_size[1]/2))
+
+### Initiallizes the game environment
+pygame.init()
+screen = pygame.display.set_mode(screen_size)
+pygame.display.set_caption('Head Movement')
+clock = pygame.time.Clock()
+
+### Directions
+STAY = (0,0)
+UP = (0,-1)
+DOWN = (0,1)
+LEFT = (-1,0)
+RIGHT = (1,0)
+
+### Colors
+BLACK = (0,0,0)
+GREEN = (0,180,0)
+WHITE = (255,255,255)
+RED = (255, 0, 0)
+DARK_GREEN = (0,100,0)
+
+### Adds player(s)
+p1 = Player((360,240),STAY)
+p2 = Player((280,240),STAY)
+p2.body_color = GREEN
+p2.head_color = DARK_GREEN
+
+### Position of fruit
+x_fruit = random.randrange(20, 620, 20)
+y_fruit = random.randrange(20, 460, 20)
+fruit = (x_fruit, y_fruit)
+
+### Adding font and size for text in game
+font = pygame.font.SysFont("Lucida Sans Typewriter", 50)
+
+### Setting initial speed
+speed = 3
+
+### Runtime script
+running = True
+while running:
+
+    clock.tick(speed) # Controls play speed
+    screen.fill(BLACK)
+
+    # Game over if player runs into their own body
+    if (p1.pos) in p1.body:
+        final_screen()
+
+    if (p2.pos) in p2.body:
+        final_screen()
+
+    # If player eat fruit, body will increase by one and new fruit will be randomed out
+    # When body lenght increases, increase speed
+    if (p1.pos) == fruit:
+        p1.grow()
+        x_fruit = random.randrange(20, 620, 20)
+        y_fruit = random.randrange(20, 460, 20)
+        fruit = (x_fruit, y_fruit)
+        if len(p1.body) in (2,4,7,10):
+            speed = speed + 1
+
+    if (p2.pos) == fruit:
+        p2.grow()
+        x_fruit = random.randrange(20, 620, 20)
+        y_fruit = random.randrange(20, 460, 20)
+        fruit = (x_fruit, y_fruit)
+        if len(p2.body) in (2,4,7,10):
+            speed = speed + 1
+
```

Comment:
There's some repetition for `p1` and `p2`. This is fragile – it's easy to modify one occurrence, and not the other. Think about how to eliminate this, for example `for p in (p1, p2):`.